### PR TITLE
Remove MasterStorageTierAssoc and WorkerStorageTierAssoc

### PR DESCRIPTION
**Pull request by rongrong**
### What changes are proposed in this pull request?

Remove MasterStorageTierAssoc and WorkerStorageTierAssoc, change AbstractStorageTierAssoc
to non-abstract and use for both. Add two static object to replace the parameter-free constructors.

### Why are the changes needed?

The two classes are the same except the default constructor initializes the underlying abstract class
with different parameters. The classes does not hold any data or logic so they don't need to exist.

### CODE REVIEW

File: core/server/common/src/main/java/alluxio/StorageTierAssocImplementation.java [hunk 1]
```diff
@@ -28,7 +28,8 @@
  * {@link ServerConfiguration}.
  */
 @ThreadSafe
-public abstract class AbstractStorageTierAssoc implements StorageTierAssoc {
+public final class StorageTierAssocImplementation
```
Comment (jiacheliu3) [inline core/server/common/src/main/java/alluxio/StorageTierAssocImplementation.java:31]: nit: In the current code I don't see pattern like using interface `Foo` and implementation class `FooImplementation`. It's more frequent that we use `interface Foo` then `class DefaultFoo implements Foo`. I suggest keeping the naming consistent.

Taking a step back, if an interface only has one implementation and there's no obvious need to extend that, I think it's fair to just remove that interface and only keep the implementation class. I'm fine with that if that makes your naming easier.

File: core/server/master/src/main/java/alluxio/master/meta/AlluxioMasterRestServiceHandler.java [hunk 2]
```diff
@@ -1121,19 +1119,6 @@ public Response getWebUIMetrics() {
     }, ServerConfiguration.global());
   }
 
-  private Comparator<String> getTierAliasComparator() {
```
Comment (jiacheliu3) [inline core/server/master/src/main/java/alluxio/master/meta/AlluxioMasterRestServiceHandler.java:1124]: just a sanity check, is this not used anywhere?

**Review by jiacheliu3 — APPROVED** [review]
LGTM, thanks for the refactor! Left some comments regarding the format and naming conventions.

**Review by jja725 — APPROVED** [review]
LGTM excpet the naming of StorageTierAssocImplementation

File: core/server/master/src/main/java/alluxio/master/meta/AlluxioMasterRestServiceHandler.java (hunk 2 above)
Comment (rongrong) [reply core/server/master/src/main/java/alluxio/master/meta/AlluxioMasterRestServiceHandler.java:1124]: It was grey according to IntelliJ. Consider that this is a private function, I'd trust if it's not called in this file, it's not used?

File: core/server/common/src/main/java/alluxio/StorageTierAssocImplementation.java (hunk 1 above)
Comment (rongrong) [reply core/server/common/src/main/java/alluxio/StorageTierAssocImplementation.java:31]: Ya, I hate the name `DefaultXXX`, makes no sense to me. I'd remove the interface if it's not in a separate package. Basically the interface is in common but the implementation is not. Eventually I'd like to restructure the packaging so we can remove the interface as well.

**Comment by rongrong** [comment]
alluxio-bot, merge this please

**Comment by jiacheliu3** [comment]
alluxio-bot, cherry-pick this to branch-2.8 please

**Pull request merged**